### ISSUE TITLE
Publishing Azure Website fails

### ISSUE THREAD

**Issue by cwoolum** [Microsoft/vso-agent-tasks#693]
When trying to deploy an Azure website, i am getting an exception that the site is already created and cannot be created again. This deploy process was working until October 23rd. Did anything change in the deploy script? I think it thinks the site doesn't exist and tries to create it again.

```
2015-10-29T03:19:24.6803000Z Get-AzureWebSite -Name WaxonService -ErrorAction SilentlyContinue
 
2015-10-29T03:19:31.7421204Z New-AzureWebSite -Name WaxonService -Location South Central US
 
2015-10-29T03:19:34.9330598Z ##[error]Conflict: Website with given name WaxonService already exists.
 
2015-10-29T03:19:34.9350596Z Publish-AzureWebsiteProject -Name "WaxonService" -Package "C:\a\1\a\Waxon.Service.zip" 
 
2015-10-29T03:19:51.0621612Z Finishing task: AzureWebPowerShellDeployment
```

**Comment by bryanmacfarlane** [comment]
assigning to yaswanth to investigate

**Comment by yaananth** [comment]
@cwoolum There has been no changes in the way the script runs.
It is quite possible that Get-AzureWebsite returns some error while retrieving the website and since we silently eat up the error there is no way for us to know.
However, this will change in M90 deployment, as we capture the error and print it out as a warning.
Could you try the command - `Get-AzureWebSite -Name WaxonService` on azure powershell commandprompt and see if it throws out some error?

Last time we saw this, it was because the website has VS2015 debugging mode on, I am not sure if that's fixed yet, but the workaround was to turn it off and use VS2013 instead at the moment,

**Comment by cwoolum** [comment]
Debugging was not enabled but when I had used it previously, it was set as 2015.

To Fix:
1. Enable Debugging and set version to 2013
2. Save changes
3. Disable debugging and Save Changes

Go figure... haha

**Comment by ianschmitz** [comment]
This issue still exists. Maybe it should be re-opened?

When I was trying to publish from VSO after using remote debugging, i got the error in the VSO build console:
```
2015-11-09T22:51:58.9878309Z ##[warning]System.ArgumentException: Requested value 'VS2015' was not found.
2015-11-09T22:51:58.9888361Z ##[warning]   at Microsoft.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2015-11-09T22:51:58.9898314Z ##[warning]   at Microsoft.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccess(Task task)
2015-11-09T22:51:58.9898314Z ##[warning]   at Microsoft.WindowsAzure.Management.WebSites.WebSiteOperationsExtensions.GetConfiguration(IWebSiteOperations operations, String webSpaceName, String webSiteName)
2015-11-09T22:51:58.9908303Z ##[warning]   at Microsoft.WindowsAzure.Commands.Utilities.Websites.WebsitesClient.GetWebsiteConfiguration(String name)
2015-11-09T22:51:58.9908303Z ##[warning]   at Microsoft.WindowsAzure.Commands.Websites.GetAzureWebsiteCommand.WriteWebsite(Site websiteObject)
2015-11-09T22:51:58.9918302Z ##[warning]   at Microsoft.WindowsAzure.Commands.Websites.GetAzureWebsiteCommand.<GetByName>b__0()
2015-11-09T22:51:58.9928294Z ##[warning]   at Microsoft.WindowsAzure.Commands.Websites.GetAzureWebsiteCommand.Do(Action call)
2015-11-09T22:51:58.9928294Z ##[warning]   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
2015-11-09T22:51:58.9948296Z New-AzureWebSite -Name fakesite -Location West US
2015-11-09T22:52:01.7199514Z ##[error]Conflict: Website with given name trackmeet-webapi-test already exists.
```

The only way i could fix it was to change the `Remote Visual Studio version` inside of the Azure Portal to `2013` instead of `2015`. Not sure on the implications of changing this but I was able to publish in the mean time.

**Comment by cwoolum** [comment]
It's not a VSO issue... It's a problem with the Powershell Command Get-AzureWebSite. Not sure who handles that....

**Comment by ramedey** [comment]
Today I've experienced the same issue. If it is not a VSO issue, where can I report it?

**Comment by yaananth** [comment]
Probably here - https://github.com/Azure/azure-powershell

**Comment by ramedey** [comment]
Thanks

**Comment by yaananth** [comment]
Infact there is an active issue https://github.com/Azure/azure-powershell/issues/462

**Comment by mxa0079** [comment]
It looks like there is a fix coming based on the conversation referenced by @yaananth Could we reopen this issue to track it until the fix makes it to production?

BTW, the workaround mentioned by @cwoolum is not working for me.

**Comment by kpgeek** [comment]
I got the similar issue. I tried to implement the steps as suggested by 
@cwoolum. No luck :( I am getting the same error.  

**Detail Error:**
**Get-AzureWebSite -Name dummy123 -ErrorAction SilentlyContinue -ErrorVariable azureWebSiteError -Slot production
****[warning]System.ArgumentException: Requested value 'Dynamic' was not found.**
[warning]   at Microsoft.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[warning]   at Microsoft.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccess(Task task)
[warning]   at Microsoft.WindowsAzure.Management.WebSites.WebSpaceOperationsExtensions.ListWebSites(IWebSpaceOperations operations, String webSpaceName, WebSiteListParameters parameters)
[warning]   at Microsoft.WindowsAzure.Commands.Utilities.Websites.Services.WebSitesManagementExtensionMethods.GetFromAzure(IWebSiteManagementClient client, String website)
[warning]   at Microsoft.WindowsAzure.Commands.Utilities.Websites.WebsitesClient.GetWebsite(String name)
[warning]   at Microsoft.WindowsAzure.Commands.Utilities.Websites.WebsitesClient.GetWebsiteConfiguration(String name)
[warning]   at Microsoft.WindowsAzure.Commands.Websites.GetAzureWebsiteCommand.WriteWebsite(Site websiteObject)
[warning]   at Microsoft.WindowsAzure.Commands.Websites.GetAzureWebsiteCommand.<GetByName>b__0()
[warning]   at Microsoft.WindowsAzure.Commands.Websites.GetAzureWebsiteCommand.Do(Action call)
[warning]   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
**New-AzureWebSite -Name dummy123 -Location West US -Slot production
Conflict: Website with given name dummy123 already exists.**
Publish-AzureWebsiteProject -Name "dummy123" -Package "C:\a\00b3423458\dummy123\drop\dummy123.zip" -Slot "production"  -ErrorVariable publishAzureWebsiteError
[warning]Cannot get website, deployment status is not updated

**Steps followed:**

**1. Enabled remote debugging ON, Set VS Version to 2013 and saved the changes**

![image](https://cloud.githubusercontent.com/assets/12850067/14396122/9ce6ba48-fd8a-11e5-8ba7-d36697ea7df0.png)

**2. Disabled remote debugging OFF and saved the changes. Now, I see VS Version 2013**

![image](https://cloud.githubusercontent.com/assets/12850067/14396592/554c3ca0-fd8d-11e5-9793-556f06b21261.png)

3. After both steps, I tried to redeploy website. Its giving the same error. I am wondering why Get-AzureWebsite command is unable to recognize whether webapp is existed or not in the portal. Why New-AzureWebSite -Name command is adding in the script? Can we customize the "Azure web deployment task" ps script? 

**I would really appreciate if any solution / workaround for this issue. **

**Comment by RoopeshNair** [comment]
@kpgeek 

Based on the update here https://connect.microsoft.com/VisualStudio/feedback/details/2559822/-error-requested-value-dynamic-was-not-found 

Removing Azure Functions solves the problem

**Comment by PirateBread** [comment]
Haha this is hilarious. Adding a functions app to your subscription breaks the AzureWebsite command.

So you add a function on a test resource group to play around with things and suddenly your project deployments are breaking because the AzureWebsite command starts throwing errors.

**Comment by ddobric** [comment]
Hi all,

I have same issue when creating a new fresh website. Is this fixed? Is this production ready?

**Comment by mohdumairakbar** [comment]
Hi All, 

I am facing the same issue "Requested value 'Dynamic' was not found." while running New-AzureWebsiteJob  on build definition through powershell script.

This was working fine few days back. I got this error and now stuck on it.

Any help. Highly Appreciated.

Thanks
Umair

**Comment by ryanbennett** [comment]
I received the same error after creating an Azure Function and trying to deploy my website.

**Comment by mohdumairakbar** [comment]
Installing the latest version of Powershell through web platfrom installer worked for me.

Make sure you restart your machine after installing latest powershell else you will face error "he required module 'Azure.Storage' is not loaded"

**Comment by ryanbennett** [comment]
Deleting the Azure function from my App Services screen let me continue building. Thankfully I was just testing the Function capabilities and not using it for anything

**Comment by leelep** [comment]
I experienced the same issue using the new DevOps web site, and it was due to being logged in to DevOps with the wrong account.  The DevOps site doesn't seem to recognize the difference between work and personal accounts of the same email address.  The Get-AzureWebSites PowerShell command does distinguish between the two.  If you have multiple accounts and experience the described issue then double check that you are signed in to DevOps with an account that has permissions for the Azure web site.
